gammapy/background/off_data_background_maker.py

```diff
@@ -0,0 +1,248 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+from __future__ import absolute_import, division, print_function, unicode_literals
+import os
+import logging
+import numpy as np
+from astropy.table import Table
+import astropy.units as u
+from astropy.table import join as table_join
+from ..data import ObservationTable, ObservationGroupAxis, ObservationGroups
+from .models import CubeBackgroundModel
+from .models import EnergyOffsetBackgroundModel
+from ..utils.energy import EnergyBounds
+from ..utils.axis import sqrt_space
+from ..extern.pathlib import Path
+
+__all__ = [
+    'OffDataBackgroundMaker',
+]
+
+log = logging.getLogger(__name__)
+
+
+class OffDataBackgroundMaker(object):
+    def __init__(self, data_store, outdir=None, run_list=None, obs_table_grouped_filename=None,
+                 group_table_filename=None):
+        """OffDataBackgroundMaker class.
+
+        Class that will select an OFF list run from a Data list and then group this runlist in group of
+        zenithal angle and efficiency. Then for each group, it will compute the background rate model in
+        3D *(X, Y, energy)* or 2D *(energy, offset)* via the class `~gammapy.background.CubeBackgroundModel` (3D) or
+         `~gammapy.background.EnergyOffsetBackgroundModel` (2D).
+
+        Parameters
+        ----------
+        data_store : `~gammapy.data.DataStore`
+            Data for the background model
+        run_list : str
+            filename where is store the OFF run list
+        outdir : str
+            directory where will go the output
+        obs_table_grouped_filename : str
```

Comment:
Is there value in having `obs_table_grouped_filename` and `group_table_filename` as parameters here?
Why not just store the default in `__init__`, and if the user really wants to customize those,
then they can do this:
```
maker = OffDataBackgroundMaker()
maker. obs_table_grouped_filename = 'my_custom_name'
```